DirectProgramming/C++SYCL_FPGA/Tutorials/Features/multi_aocx/device_code_split/README.md

```diff
@@ -0,0 +1,259 @@
+# Using the flows test 'device_code_split'
+
+This FPGA tutorial demonstrates the -fsycl-device-code-split=per_source|per_kernel
+
+| Optimized for                     | Description
+|:---                               |:---
+| OS                                | Linux* Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE* 15 <br> Windows* 10
+| Hardware                          | Intel&reg; Programmable Acceleration Card (PAC) with Intel Arria&reg; 10 GX FPGA <br> Intel&reg; FPGA Programmable Acceleration Card (PAC) D5005 (with Intel Stratix&reg; 10 SX) <br> Intel&reg; FPGA 3rd party / custom platforms with oneAPI support <br> **Note**: Intel&reg; FPGA PAC hardware is only compatible with Ubuntu 18.04*
+| Software                          | Intel® oneAPI DPC++/C++ Compiler
+| What you will learn               | How to use the `-fsycl-device-code-split=per_source|per_kernel` to create multiple aocx.
+| Time to complete                  | 30 minutes
+
+> **Note**: Even though the Intel DPC++/C++ OneAPI compiler is enough to compile for emulation, generating reports and generating RTL, there are extra software requirements for the simulation flow and FPGA compiles.
+>
+> For using the simulator flow, one of the following simulators must be installed and accessible through your PATH:
+> - Questa*-Intel® FPGA Edition
+> - Questa*-Intel® FPGA Starter Edition
+> - ModelSim® SE
+>
+> When using the hardware compile flow, Intel® Quartus® Prime Pro Edition must be installed and accessible through your PATH.
+
+## Purpose
+The purpose of this tutorial is to describe the `-fsycl-device-code-split` flag and to show how it can be leveraged to break down the compilation of several kernels into smaller individual compiles.
+
+The tutorial will also describe the `-reuse-exe=<exe>` flag and show how it can enable rapid recompilation of host-only code changes.
```

Comment:
````suggestion
This sample is an FPGA tutorial that demonstrates the `-fsycl-device-code-split=per_source|per_kernel`

| Area                 | Description
|:--                      |:--
| What you will learn  | How to use the `-fsycl-device-code-split=per_source|per_kernel` compiler flag to create multiple aocx
| Time to complete     | 20 minutes
| Category             | Concepts and Functionality

## Purpose
This FPGA tutorial shows how to use the `-fsycl-device-code-split` compiler flag and how it can be leveraged to break down the compilation of several kernels into smaller individual compiles.

The tutorial will also describe the `-reuse-exe=<exe>` flag and show how it can enable rapid recompilation of host-only code changes.

## Prerequisites

| Optimized for       | Description
|:---                         |:---
| OS                        | Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE* 15 <br> Windows* 10
| Hardware            | Intel® Agilex® 7, Arria® 10, and Stratix® 10 FPGAs
| Software            | Intel® oneAPI DPC++/C++ Compiler

> **Note**: Even though the Intel DPC++/C++ OneAPI compiler is enough to compile for emulation, generating reports and generating RTL, there are extra software requirements for the simulation flow and FPGA compiles.
>
> For using the simulator flow, Intel® Quartus® Prime Pro Edition and one of the following simulators must be installed and accessible through your PATH:
> - Questa*-Intel® FPGA Edition
> - Questa*-Intel® FPGA Starter Edition
> - ModelSim® SE
>
> When using the hardware compile flow, Intel® Quartus® Prime Pro Edition must be installed and accessible through your PATH.

> **Warning** Make sure you add the device files associated with the FPGA that you are targeting to your Intel® Quartus® Prime installation.

This sample is part of the FPGA code samples.
It is categorized as a Tier 2 sample that demonstrates a compiler feature.

```mermaid
flowchart LR
   tier1("Tier 1: Get Started")
   tier2("Tier 2: Explore the Fundamentals")
   tier3("Tier 3: Explore the Advanced Techniques")
   tier4("Tier 4: Explore the Reference Designs")

   tier1 --> tier2 --> tier3 --> tier4

   style tier1 fill:#0071c1,stroke:#0071c1,stroke-width:1px,color:#fff
   style tier2 fill:#f96,stroke:#333,stroke-width:1px,color:#fff
   style tier3 fill:#0071c1,stroke:#0071c1,stroke-width:1px,color:#fff
   style tier4 fill:#0071c1,stroke:#0071c1,stroke-width:1px,color:#fff
```
````